### PR TITLE
Fixes #3699 Deadfire LRMs and SRMs have incorrect range brackets

### LINKED ISSUE #3699: Deadfire LRMs and SRMs have incorrect range brackets 0.49.8
I don't have my copy of IO on me, but at least according to Sarna.net, the deadfire LRMs and SRMs in MegaMek have the wrong range brackets.  MegaMek has the deadfire missiles as having the same range brackets and minimum range penalties as the regular missiles, while Sarna shows the following range brackets:

image

### PR DESCRIPTION
**Pull request by HammerGS**
With lots of help from Juliez (who has also reviewed and tested) I've got Dead Fire missiles implemented with the correct ranges and updates to display. 

One area I noticed we are missing is the correct BV for them but it is one of the weapons where every Ammo size has a different BV. I've added a TODO for this.

### CODE REVIEW

File: megamek/src/megamek/common/AmmoType.java [hunk 1]
```diff
@@ -13432,13 +13437,15 @@ public AmmoType createMunitionType(AmmoType base) {
                     || (munition.getAmmoType() == AmmoType.T_NLRM))
                     && (munition.getMunitionType() == AmmoType.M_AX_HEAD)) {
                 cost *= 0.5;
+                bv *= 1;
```
Comment (Windchild292) [inline megamek/src/megamek/common/AmmoType.java:13440]: multiplying by 1 doesn't do anything?

File: megamek/src/megamek/common/AmmoType.java [hunk 2]
```diff
@@ -13432,13 +13437,15 @@ public AmmoType createMunitionType(AmmoType base) {
...
             }
 
             if (((munition.getAmmoType() == AmmoType.T_LRM) || (munition.getAmmoType() == AmmoType.T_LRM_IMP)
                     || (munition.getAmmoType() == AmmoType.T_MML) || (munition.getAmmoType() == AmmoType.T_SRM)
                     || (munition.getAmmoType() == AmmoType.T_SRM_IMP) || (munition.getAmmoType() == AmmoType.T_NLRM))
                     && (munition.getMunitionType() == AmmoType.M_SMOKE_WARHEAD)) {
                 cost *= 0.5;
+                bv *= 1;
```
Comment (Windchild292) [inline megamek/src/megamek/common/AmmoType.java:13448]: multiplying by 1 doesn't do anything?

File: megamek/src/megamek/common/AmmoType.java [hunk 3]
```diff
@@ -13457,12 +13464,14 @@ public AmmoType createMunitionType(AmmoType base) {
                     || (munition.getAmmoType() == AmmoType.T_MML) || (munition.getAmmoType() == AmmoType.T_NLRM))
                     && (munition.getMunitionType() == AmmoType.M_SEMIGUIDED)) {
                 cost *= 3;
+                bv *= 1;
```
Comment (Windchild292) [inline megamek/src/megamek/common/AmmoType.java:13467]: multiplying by 1 doesn't do anything?

File: megamek/src/megamek/common/AmmoType.java [hunk 4]
```diff
@@ -13457,12 +13464,14 @@ public AmmoType createMunitionType(AmmoType base) {
...
             }
 
             if (((munition.getAmmoType() == AmmoType.T_LRM) || (munition.getAmmoType() == AmmoType.T_LRM_IMP)
                     || (munition.getAmmoType() == AmmoType.T_MML) || (munition.getAmmoType() == AmmoType.T_NLRM))
                     && (munition.getMunitionType() == AmmoType.M_SWARM)) {
                 cost *= 2;
+                bv *= 1;
```
Comment (Windchild292) [inline megamek/src/megamek/common/AmmoType.java:13474]: multiplying by 1 doesn't do anything?

File: megamek/src/megamek/common/AmmoType.java [hunk 5]
```diff
@@ -13551,11 +13560,18 @@ public AmmoType createMunitionType(AmmoType base) {
...
                             || (munition.getMunitionType() == AmmoType.M_FRAGMENTATION))) {
                 cost *= 2;
             }
 
+            if (((munition.getAmmoType() == AmmoType.T_LRM) || (munition.getAmmoType() == AmmoType.T_LRM_IMP)
+                    || (munition.getAmmoType() == AmmoType.T_MML) || (munition.getAmmoType() == AmmoType.T_SRM)
+                    || (munition.getAmmoType() == AmmoType.T_SRM_IMP) || (munition.getAmmoType() == AmmoType.T_NLRM))
+                    && ((munition.getMunitionType() == AmmoType.M_DEAD_FIRE))) {
+                cost *= .6;
+                //TODO - DEAD-FIRE AMMO needs BV which is not a constant but launcher Ammo.
```
Comment (Windchild292) [inline megamek/src/megamek/common/AmmoType.java:13572]:
```suggestion
                cost *= 0.6;
                // TODO - DEAD-FIRE AMMO needs BV which is not a constant but launcher Ammo. 
```

**Review by Windchild292 — COMMENTED** [review]
A few comments about lines that don't do anything, and one pair of readability issues (you always want to have a leading zero before a decimal to prevent misreads, and a space at the start of a text line)

File: megamek/src/megamek/common/AmmoType.java (hunk 1 above)
Comment (HammerGS) [reply megamek/src/megamek/common/AmmoType.java:13440]: Reminder that the BV has been checked. Beats have a blank line with no idea that it's been compared to source.

File: megamek/src/megamek/common/AmmoType.java (hunk 2 above)
Comment (HammerGS) [reply megamek/src/megamek/common/AmmoType.java:13448]: Reminder that the BV has been checked. Beats have a blank line with no idea that it's been compared to source.

File: megamek/src/megamek/common/AmmoType.java (hunk 3 above)
Comment (HammerGS) [reply megamek/src/megamek/common/AmmoType.java:13467]: Reminder that the BV has been checked. Beats have a blank line with no idea that it's been compared to source.

File: megamek/src/megamek/common/AmmoType.java (hunk 4 above)
Comment (HammerGS) [reply megamek/src/megamek/common/AmmoType.java:13474]: Reminder that the BV has been checked. Beats have a blank line with no idea that it's been compared to source.

File: megamek/src/megamek/common/AmmoType.java (hunk 1 above)
Comment (Windchild292) [reply megamek/src/megamek/common/AmmoType.java:13440]: Can we comment it out instead then?

**Review by Windchild292 — APPROVED** [review]

File: megamek/src/megamek/common/AmmoType.java (hunk 4 above)
Comment (HammerGS) [reply megamek/src/megamek/common/AmmoType.java:13474]: For be its easier to see the missing BV formula than read the comment for everyone. If I know they are missing a BV multiplier then I know we have an issue of some type.

**Pull request merged**